hw14/election.py

```diff
@@ -0,0 +1,336 @@
+# Name: Samantha Prince
+# CodeFellows Python F2: SEA-C42
+# Homework 14: Election prediction
+
+import csv
+import os
+import time
+
+
+def read_csv(path):
+    """
+    Reads the CSV file at path, and returns a list of rows. Each row is a
+    dictionary that maps a column name to a value in that column, as a string.
+    """
+    output = []
+    for row in csv.DictReader(open(path)):
+        output.append(row)
+    return output
+
+
+###############################################################################
+# Problem 1: State edges
+###############################################################################
+
+def row_to_edge(row):
+    """
+    Given an *ElectionDataRow* or *PollDataRow*, returns the
+    Democratic *Edge* in that *State*.
+    """
+    return float(row["Dem"]) - float(row["Rep"])
+
+
+def state_edges(election_result_rows):
+    """
+    Given a list of *ElectionDataRow*s, returns *StateEdge*s.
+    The input list has no duplicate *States*;
+    that is, each *State* is represented at most once in the input list.
+    """
+# my original answer
+    # d = {}
+    # for row in election_result_rows:
+        # state = row['State']
+        # dem = row['Dem']
+        # rep = row['Rep']
+        # d[state] = float(dem) - float(rep)
+    # return d
```

Comment:
When merging branches and doing pull requests, it's best to just delete old and commented out code. As long as you included it in your commit history at some point, you can always find it again. Clean code is best code.